lib/device-log/ios-log.js

```diff
@@ -249,7 +252,25 @@ class IOSLog {
   async getAllLogs () {
     return this.logs;
   }
+
+  /**
+   * Get 'idevicesyslog' subprocess for a given udid.
+   * 
+   * This function caches and re-uses instances so that there's
+   * just one instance per device (cache by UDID)
+   */
+  static async _getIDeviceSysLogProcess (cmd, args, env, udid) {
+    if (!IOSLog.cachedIDeviceSysLogs[cmd]) {
+      IOSLog.cachedIDeviceSysLogs[cmd] = {};
+    }
+    if (!IOSLog.cachedIDeviceSysLogs[cmd][udid]) {
+      IOSLog.cachedIDeviceSysLogs[cmd][udid] = new SubProcess(cmd, args, {env});
+    }
+    return IOSLog.cachedIDeviceSysLogs[cmd][udid];
```

Comment:
here we don't actually check if the process is running even if it is in the cache. In general, it should be removed from the cache as soon as it is terminated.

Reply:
Makes sense. I should bind a method to the `exit` event that deletes the reference to the SubProcess?